container/docker/factory.go

```diff
@@ -204,6 +208,35 @@ func parseDockerVersion(full_version_string string) ([]int, error) {
 	return version_array, nil
 }
 
+func getStorageDir(dockerInfo *docker.Env) (string, error) {
+	storageDriverInfo := dockerInfo.GetList("DriverStatus")
+	if len(storageDriverInfo) == 0 {
+		return "", fmt.Errorf("failed to find docker storage driver options")
+	}
+
+	entries := map[string]string{}
+	for _, data := range storageDriverInfo {
+		if data == "" {
+			continue
+		}
+		var parts [][]string
```

Comment:
Since we are just looking for one element ("Root Dir") we don't need to populate the map, and could just return when we find the entry we are looking for.  It is a pretty minor optimization, and only worth doing if it would improve the clarity of the code